src/software/ai/navigator/path_planner/hrvo/hrvo_simulator.cpp

```diff
@@ -0,0 +1,262 @@
+#include "hrvo_simulator.h"
+
+HRVOSimulator::HRVOSimulator(const RobotConstants_t &robot_constants,
+                             TeamColour friendly_team_colour)
+    : robot_constants(robot_constants),
+      robots(),
+      world(std::nullopt),
+      primitive_set(),
+      friendly_team_colour(friendly_team_colour)
+{
+}
+
+void HRVOSimulator::updateWorld(const World &world, double time_step)
+{
+    this->world = world;
+    // TODO (#2498): Update implementation to correctly support adding and removing agents
+    //               to represent the newly added and removed friendly/enemy robots in the
+    //               World.
+    robots.clear();
+    for (const Robot &friendly_robot : world.friendlyTeam().getAllRobots())
+    {
+        configureHRVORobot(friendly_robot, time_step);
+    }
+
+    for (const Robot &enemy_robot : world.enemyTeam().getAllRobots())
+    {
+        configureLVRobot(enemy_robot, time_step);
+    }
+}
+
+
+void HRVOSimulator::updatePrimitiveSet(const TbotsProto::PrimitiveSet &new_primitive_set, double time_step)
+{
+    primitive_set = new_primitive_set;
+
+    // Update all friendly agent's primitives
+    for (auto &[robot_id, primitive] : primitive_set.robot_primitives())
+    {
+        auto friendly_robot = robots.find(robot_id);
+        if (friendly_robot == robots.end())
+        {
+            return;
+        }
```

Comment:
We should continue to next iteration if an Agent doesn't exist with a matching robot id. Could probably combine it with the next if statement